client/shared/src/main/scala/org/http4s/client/oauth1/SignatureAlgorithm.scala

```diff
@@ -70,30 +68,51 @@ trait SignatureAlgorithm {
     */
   def name: String
 
+  @deprecated("Use generateBase64[F[_]: MonadThrow] instead", "0.23.28")
+  def generate[F[_]: MonadThrow](input: String, secretKey: String): F[String] =
+    MonadThrow[F].catchNonFatal(generate(input, secretKey))
+
+  @deprecated("Use generate[F[_]: MonadThrow] instead", "0.22.5")
+  def generate(input: String, secretKey: String): String =
+    generate[SyncIO](input, secretKey).unsafeRunSync()
```

Comment:
Oh Lord, it turns out that we've been living with a StackOverflow time bomb in the public API all these years. Hardly anyone has tried to go down this path, but still!